lesson16/script.js

```diff
@@ -0,0 +1,188 @@
+}
+
+const renderTopics = (result) => {
+    const topicsArea     = document.getElementById('js-topics-list');
+    const fragmentTopics = document.createDocumentFragment();
+    for (const topics of result){
+        const category     = document.createElement('li');
+        const categroyLink = document.createElement('a');
+        category.classList.add('topics-list__item');
+        categroyLink.textContent = topics.categories;
+        fragmentTopics.appendChild(category).appendChild(categroyLink);
+    }
+    topicsArea.appendChild(fragmentTopics);
+}
+
+const renderImages = (result) => {
+    const contentsArea  = document.getElementById('js-contents-area');
+    const imageArea     = document.createElement('div');
+    imageArea.className = "contents-item topics-img";
+
+    const fragmentTopicImg = document.createDocumentFragment();
+    for (const topics of result){
+        const topicsImg = document.createElement('img');
+        topicsImg.id    = `${topics.categories + '-img' }`;
+        topicsImg.src   = topics.picture;
+        topicsImg.classList.add('topics-img__item');
+        fragmentTopicImg.appendChild(topicsImg);
+    }
+    contentsArea.appendChild(imageArea).appendChild(fragmentTopicImg);
+}
+
+const showNewIcon = () => {
+    const newIcon       = document.createElement('span');
+    newIcon.className   = "new-icon";
+    newIcon.textContent = "New";
+    return newIcon;
+}
+
+const showCommentIcon = () => {
+    const commentIcon     = document.createElement('img');
+    commentIcon.src       = "img/comment-dots-solid.svg";
+    commentIcon.className = "comment-icon";
+    return commentIcon;
+}
+
+const showCommentNumber = (commentLength) => {
+    const commentNumber       = document.createElement('span');
+    commentNumber.textContent = commentLength;
+    commentNumber.className   = "comment-number";
+    return commentNumber;
+}
+
+const renderArticles = (result) => {
+    const contentsArea    = document.getElementById('js-contents-area');
+    const articleArea     = document.createElement('div');
+    articleArea.className = "contents-item articles";
+
+    const fragmentAritcle = document.createDocumentFragment(); 
+    for (const topics of result){
+        const articleList = document.createElement('ul');
+        articleList.id    = `${topics.categories + "-articles"}`;
+        articleList.classList.add('article-list');
+
+        const articles = topics.articles;
+        for(const article of articles ){
+            const articleItem   = document.createElement('li');
+            const articleLink   = document.createElement('a'); 
+            articleItem.classList.add('article-list__item');
+            articleLink.textContent  = article.title;
+
+            const commentLength = article.comment.length; 
+            const articleDateMs = new Date(article.date).getTime();
+            const periodDayMs   = new Date().getTime() - (3*24*60*60*1000);
```

Comment:
IMO
How about storing 3*24*60*60*1000 in a variable? I thought it might be difficult to imagine what these numbers were. Now It is a magic number.

Reply:
IMO:
moriken-sensei said that there is a library API for date manipulation called "date-fns".
I will put the URL for it on Discord.

Reply:
@csakyo 
It's certainly hard to understand!
I'll put it in a variable.

Reply:
@yasu-pro 
Thank you for your comments and useful information.
I will find out about data-fns!

Reply:
@yasu-pro @csakyo 
I used date-fns to make the process functional.
https://github.com/nakagawamai/js-lesson/pull/18/commits/a774522f6d525dace713a4a23b5ca3b6a08407cb

Reply:
@yasu-pro @csakyo 
I changed to install date-fns with npm package.
https://github.com/nakagawamai/js-lesson/pull/18/commits/e47055a1a6e743674707f277b1695d239cd69ef2